craft_parts/state_manager/states.py

```diff
@@ -0,0 +1,67 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2016-2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Helpers and definitions for lifecycle states."""
+
+import logging
+from pathlib import Path
+from typing import Optional
+
+import yaml
+
+from craft_parts.parts import Part
+from craft_parts.steps import Step
+
+from .build_state import BuildState
+from .prime_state import PrimeState
+from .pull_state import PullState
+from .stage_state import StageState
+from .step_state import StepState
+
+logger = logging.getLogger(__name__)
+
+
+def load_state(part: Part, step: Step) -> Optional[StepState]:
```

Comment:
Perhaps `load(cls, state_file: pathlib.Path)` should be a classmethod in StepState to keep it together?

Reply:
fwiw I don't care what the names are, but to standardize a bit we should choose something like we have for marshal/unmarshal.